components/constellation/constellation.rs

```diff
@@ -359,6 +360,30 @@ enum ExitPipelineMode {
     Force,
 }
 
+/// A script channel, that closes the script thread down when it is dropped
+pub struct ScriptChan {
+    chan: IpcSender<ConstellationControlMsg>,
+    dont_send_or_sync: PhantomData<Rc<()>>,
+}
```

Comment:
The `dont_send_or_sync` isn't necessary, it's just there to make sure that any messages that are sent on this channel come from the constellation thread. I could just remove it.